01-JS-DOM-APIs/exercise1/js/main.js

```diff
@@ -4,6 +4,10 @@ function fadein(section) {
   }
 }
 
+function fadeinSection(section) {
```

Comment:
A callback is a function refered, commonly passed to be called later on some event. So:
1. ` fadeinSection(helloWorldSection)`: this line calls fadeinSection with `helloWorldSection`  as argument.
2. The function executes and calls `fadein`  with `section` as argument, previously binded to the value of `helloWorldSection`
3. The function has no return, so its return `undefined`
4. `addEventListener` receive this `undefined` as second parameter

You need a function here, so a possibility could be:
```
function fadeinSection(section) {
  return function() {
    fadein(section);
  }
}
```
To return a function that calls `fadein` with `section` as argument.
Also, `bind`  (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind), returns a function and let's you add extra args to that new function.

```
fadein.bind(this, helloWorldSection)
```

Also, more quickly using ES6 arrow functions:

```
() => fadein(helloWorldSection)
```